ipfx/ephys_data_set.py

```diff
@@ -195,6 +195,13 @@ def get_stimulus_name(self, stim_code):
         if not self.ontology:
             raise ValueError("Missing stimulus ontology")
 
-        stim = self.ontology.find_one(stim_code, tag_type="code")
-        return stim.tags(tag_type="name")[0][-1]
-
+        try:
+            stim = self.ontology.find_one(stim_code, tag_type="code")
+            return stim.tags(tag_type="name")[0][-1]
+
+        except KeyError:
+            if self.validate_stim:
+                raise
```

Comment:
You should probably add an informative exception message to raise..

Reply:
The message is already raised by the calling function in the try block. So I thought it should be enough.

Reply:
I think `raise`ing like that breaks the stack trace though?

See: https://stackoverflow.com/questions/18188563/how-to-re-raise-an-exception-in-nested-try-except-blocks